Sources/ManageServer.php

```diff
@@ -555,6 +562,10 @@ function hideGlobalCookies()
 				fatal_lang_error('invalid_cookie_domain', false);
 		}
 
+		// Per spec, if samesite setting is 'none', cookies MUST be secure. Thems the rules. Else you lock everyone out...
+		if (!empty($_POST['samesiteCookies']) && ($_POST['samesiteCookies'] === 'none') && empty($_POST['secureCookies']))
+			fatal_lang_error('samesiteSecureRequired', false);
+
```

Comment:
I would honestly preferred if we "fixed" it instead of errored out.  But I'm fine with this for 2.1.

I think in future versions, we could attempt to "fix" the secure, which if we forced secure we would have to force https.  Maybe in this future version we could do an ajax request that would set the test cookie using the new settings and then issue another ajax request to see if the cookie is working.  Just a thought  here.

Reply:
I really don't like silently changing settings...   Let 'em know when they've done something wrong so they can research it better and fix it themselves.

Reply:
Yea, which we could address in a future version maybe with a hint that this setting is incompatible with another setting.  But I think again, we can address that in a future version where maybe we have a proper js/php function that can support this all natively out of the box.  The test would be an idea to ensure on something like changing the cookie settings that could break the software, that we believe (via some simple testing), that the cookie change should not break the system.  But again, future version for that since the work it requires is too much to include now.  Just my thoughts as I see this and think we could do better, if we where in the alpha/beta stages..